contracts/MerkleLib.sol

```diff
@@ -99,4 +99,57 @@ library MerkleLib {
         require(index <= 255, "Index out of bounds");
         return claimedBitMap | (1 << index % 256);
     }
+
+    function verifyNeighborhood(
+        bytes32 root,
+        bytes32[] memory leaves,
+        bytes32[] calldata proof
+    ) internal pure returns (bool) {
+        // If the length is 1 (or 0) just skip the loop, since the node should just be verified directly.
+        if (leaves.length > 1) {
+            // The algorithm below successively reduces the merkle data (via hashing) to the earlier indexes.
+            // As it goes, it replaces the indices that are no longer used with bytes32(0) so future iterations know
+            // where the end of the usable data is.
+            while (leaves[1] != bytes32(0)) {
+                // Just increments. Relies on the break statement to terminate the loop.
+                for (uint256 i = 0; ; i++) {
+                    // Store this variable since this is needed twice.
+                    bool isEven = i % 2 == 0;
+
+                    // This just detects if this index is the end of the current data. That could be the end of the
+                    // allocated array or the end of the nonzero data.
+                    if (i == leaves.length - 1 || leaves[i + 1] == bytes32(0)) {
```

Comment:
What if theres a leaf equal to `bytes32(0)` in the middle of the `leaves` array? Is this not possible?